### DISCUSSION TITLE
Unintended adding of a Paragraph when using a partial

### DISCUSSION THREAD

**Discussion by topofocus** [bridgetownrb/bridgetown#572] [Q&A] [answered]
Sorry, Me again.

I have a liquid helper, included via `{% include hieronymus %} `   in Jekyll

Created a component and copied the content into the file `hieronymus.liquid`
```
<span> some content </span>
```
and replaced `insert` with `render`

In Jekyll this inserts just the code, a word. 
In Bridgetown, its inserting the code following by "</p>".

How to migrate the helper to bridgetown?

Maybe this is a topic for your _Migration-guide_ as well?

(this behavoir can be verified in  https://github.com/topofocus/bridgetown-test)

**Comment by topofocus** [comment]
The issue is present in another context, too.

The [cookie landing page template](https://github.com/topofocus/cookie) defines a `card`-component.

```ruby
<p class="..."> <%= @text %> </p>
```
which is rendered as expected.

However
```ruby
<p class="..."> <%= markdownify @text %> </p>
```
instead of simply markdownyfying the content also inserts an paragraph:

```
<p class="mt-2 mb-4 text-gray-600">
	<p>First Card: Content for this! 
                           ( <strong>interpreted</strong> by the <code class="highlighter-rouge">markdown</code> filter)</p>
		</p>
</p>
```

Completely irritating is the behavior if not only a sentence is markdownified. 
Adding a paragraph to the input inserts a div and  and adapts the highlight-class to the added sentence.

```
<p class="mt-2 mb-4 text-gray-600">
		<p>First Card: Content for this! 
                           ( <strong>interpreted</strong> by the <code class="highlighter-rouge">markdown</code> filter)</p>

<div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>                       This should be another paragraph.
</code></pre></div></div>
		</p>
```

To reproduce: Add a sentence to `tmp_cards` in the frontmatter of src/index.md  and modify src/_components/shared/card.erb  (row 14)

should I open an Issue?

**Comment by jaredcwhite** [comment]
@topofocus Can you try running the content through the `strip` method before handing it to `markdownify`? Because any newlines in the string will end up creating paragraphs, that's just how the Markdown processor works.

**Comment by topofocus** [comment, marked as answer]
So it seems, that using a ruby-class instead of a partial solves the problem.
Its even not necessary to inherent it from `Bridgetown::Component` 

Thus replacing `render 'hieronymus` with `render Hieronymus.new`  and providing this as `hieronymus.rb`in `_components`
``` ruby
class Hieronymus
  def render_in(view_context, &block)
    <<~RR
        <span class="..."> something</span>
   RR
  end
end
```
provides the expected behaviour.

**Reply by jaredcwhite** [reply]
That's an interesting finding! Off the top of my head, I can't think of a reason why that would be…I'd have to dig into all the code involved and troubleshoot further. But glad you were able to make progress.
